src/Analyzer.BicepProcessor/BicepLocationResolver.cs

```diff
@@ -34,20 +43,69 @@ public BicepLocationResolver(TemplateContext templateContext)
         /// to find the line number of in the original template.</param>
         /// <returns>The line number of the equivalent location in the original template,
         /// or 1 if it can't be determined.</returns>
-        public int ResolveLineNumber(string pathInExpandedTemplate)
+        public SourceLocation ResolveSourceLocation(string pathInExpandedTemplate)
         {
-            var jsonLine = this.jsonLineNumberResolver.ResolveLineNumber(pathInExpandedTemplate);
+            var jsonLine = this.jsonLineNumberResolver.ResolveSourceLocation(pathInExpandedTemplate).LineNumber;
 
             // Source map line numbers from Bicep are 0-indexed
             jsonLine--;
 
-            // TODO: look for mappings in other files/modules once nested templates are supported, for now just entrypoint file
-            var entrypointFile = this.sourceMap.Entries.First(entry => entry.FilePath == this.sourceMap.Entrypoint);
-            var match = entrypointFile.SourceMap.FirstOrDefault(mapping => mapping.TargetLine == jsonLine);
+            // Find the most specific match in source map
+            var bestMatch = metadata.SourceMap.Entries
+                .Select(sourceFile =>
+                {
+                    var match = sourceFile.SourceMap.FirstOrDefault(mapping => mapping.TargetLine == jsonLine);
+                    var matchSize = match != default
+                        ? sourceFile.SourceMap.Count(mapping => mapping.SourceLine == match.SourceLine)
+                        : int.MaxValue;
+                    return (sourceFile.FilePath, match?.SourceLine, matchSize);
+                })
+                .MinBy(tuple => tuple.matchSize);
 
-            return (match != null)
-                ? match.SourceLine + 1 // convert to 1-indexing
-                : 1;
+            // default to result from JSON if no matches
+            if (!bestMatch.SourceLine.HasValue)
+            {
+                return new SourceLocation(this.EntrypointFilePath, jsonLine + 1); // convert line number back to 1-indexing
+            }
+
+            // check if match is an ARM module reference, if so return location in that template
+            var moduleMetadata = this.metadata.ModuleInfo.FirstOrDefault(info => info.FileName == bestMatch.FilePath);
+            if (moduleMetadata != default && moduleMetadata.Modules.ContainsKey(bestMatch.SourceLine.Value))
+            {
+                var modulePath = moduleMetadata.Modules[bestMatch.SourceLine.Value];
+                if (modulePath.EndsWith(".json") && File.Exists(modulePath))
+                {
+                    var template = ArmTemplateCache.GetArmTemplate(modulePath);
```

Comment:
This solution for getting line number does not work for nested ARM templates. See #307.